rpc/handlers.go

```diff
@@ -383,11 +420,26 @@ func (h *Handler) TransactionReceiptByHash(hash felt.Felt) (*TransactionReceipt,
 	}
 
 	var receiptBlockNumber *uint64
+	status := StatusAcceptedL2
+
 	if blockHash != nil {
 		receiptBlockNumber = &blockNumber
+
+		l1H, jsonErr := h.l1Head()
+		if jsonErr != nil {
+			return nil, jsonErr
+		}
+
+		if isL1Verified(*receiptBlockNumber, l1H) {
```

Comment:
nit; we can avoid dereferencing

```suggestion
		if isL1Verified(blockNumber, l1H) {
```